x-pack/plugins/enterprise_search/public/applications/shared/layout/nav.tsx

```diff
@@ -154,25 +160,29 @@ export const useEnterpriseSearchNav = (alwaysReturn = false) => {
         defaultMessage: 'Build',
       }),
     },
-    {
-      id: 'relevance',
-      items: [
-        {
-          id: 'inference_endpoints',
-          name: i18n.translate('xpack.enterpriseSearch.nav.inferenceEndpointsTitle', {
-            defaultMessage: 'Inference Endpoints',
-          }),
-          ...generateNavLink({
-            shouldNotCreateHref: true,
-            shouldShowActiveForSubroutes: true,
-            to: INFERENCE_ENDPOINTS_PLUGIN.URL + INFERENCE_ENDPOINTS_PATH,
-          }),
-        },
-      ],
-      name: i18n.translate('xpack.enterpriseSearch.nav.relevanceTitle', {
-        defaultMessage: 'Relevance',
-      }),
-    },
+    ...(hasEnterpriseLicense
```

Comment:
I wonder if it would be better to register a [KibanaFeature](https://github.com/elastic/kibana/blob/main/x-pack/plugins/features/common/kibana_feature.ts#L67) and allow it to enforce the license requirements instead of manually checking this in code.

I know we have not traditionally used KibanaFeatures, but it's something we should likely start doing.

This is something we should discuss separately (maybe in the frontend alignment mtg) and investigate how the implementation would work outside of this PR.